src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/decode/ValueDecoder.scala

```diff
@@ -105,22 +106,22 @@ private[decode] object ValueDecoder {
           value.toDouble.some.asRight
         } catch {
           case _: NumberFormatException =>
-            (key, s"Cannot parse key $key with value $value into double").asLeft
+            InvalidValue(key.toString, s"Cannot parse key $key with value $value into double").asLeft
```

Comment:
It seems that type argument is another common pattern for `InvalidValue`? `double`, `boolean` etc. On the other hand it is deriveable from `key`...

Reply:
Ignore above - just thoughts aloud.